runtime/src/plugin_ref.rs

```diff
@@ -78,6 +88,14 @@ impl<'a> AsMut<Plugin> for PluginRef<'a> {
 impl<'a> Drop for PluginRef<'a> {
     fn drop(&mut self) {
         trace!("Dropping PluginRef {}", self.id);
-        // Cleanup?
+        if self.running {
+            let plugin = self.as_mut();
+
+            // Stop timer
+            if let Err(e) = plugin.stop_timer() {
+                let id = plugin.timer_id;
+                error!("Failed to stop timeout manager for {id}: {e:?}");
+            }
```

Comment:
also related to the timeout comments above (sorry if this is redundant or not actually related :laughing:)